addon/components/bs-form/element.d.ts

```diff
@@ -0,0 +1,41 @@
+import Component from '@glimmer/component';
+
+interface BsFormElementSignature<T> {
+  Element: HTMLDivElement;
+  Args: {
+    controlType?: 'text' | 'checkbox' | 'radio' | 'switch' | 'textarea' | string;
```

Comment:
I think / hope that passing a component is supported as well. But have never tested to be honest.

```suggestion
    controlType?: 'text' | 'checkbox' | 'radio' | 'switch' | 'textarea' | string | typeof Component<unknown>;
```

Reply:
this is propably not the place to pass a component type, but rather <input type={{controlType}} input type.

Reply:
Ember Bootstrap supports custom control types. E.g. the [ember-bootstrap-power-select](https://github.com/ember-bootstrap/ember-bootstrap-power-select) addon provides an integration with [Ember Power Select](https://ember-power-select.com/) allowing it to be used as a control type.

```hbs
<BsForm as |form|>
  <form.element @controlType="power-select" />
</BsForm>
```

Current that relies on naming convention. If being passed a `power-select` control type, Ember Bootstrap tries to resolve a component `<BsForm::Element::Control::PowerSelect>`. If that does not exists it falls back to a default control type.

That lookup mechanism is not fitting well with Embroider and tree-shaking. With first class component templates passing the component directly would feel much more comfortable:

```js
import PowerSelectControl from 'ember-bootstrap-power-select';

<template>
  <BsForm as |form|>
    <form.element @controlType={{PowerSelectControl}} />
  </BsForm>
</template>
```

I think / hope that is already supported today. But have never double checked.